src/platform_linux/shmem.c

```diff
@@ -278,20 +404,62 @@ platform_shm_print_usage_stats(shminfo_usage_stats *usage)
       usage->free_bytes,
       size_str(usage->free_bytes),
       (FRACTION_ARGS(free_bytes_pct) * 100),
+      usage->bytes_freed,
+      size_str(usage->bytes_freed),
+      (FRACTION_ARGS(bytes_freed_pct) * 100),
       usage->nfrees,
       usage->nf_search_skipped,
       (usage->nfrees ? (int)((usage->nf_search_skipped * 100) / usage->nfrees)
                      : 0),
-      usage->frags_inuse_HWM,
+      usage->nfrees_eq0,
+      usage->nfrees_le32,
+      usage->nfrees_le64,
+      (FRACTION_ARGS(nf_le64_pct) * 100),
+      usage->nfrees_le128,
+      (FRACTION_ARGS(nf_le128_pct) * 100),
+      usage->nfrees_le256,
+      (FRACTION_ARGS(nf_le256_pct) * 100),
+      usage->nfrees_le512,
+      (FRACTION_ARGS(nf_le512_pct) * 100),
+      usage->nfrees_le1K,
+      (FRACTION_ARGS(nf_le1K_pct) * 100),
+      usage->nfrees_le2K,
+      (FRACTION_ARGS(nf_le2K_pct) * 100),
+      usage->nfrees_le4K,
+      (FRACTION_ARGS(nf_le4K_pct) * 100),
+      usage->nfrags_inuse,
+      usage->nfrags_inuse_HWM,
       usage->large_frags_found_in_use,
       usage->used_by_large_frags_bytes,
-      size_str(usage->used_by_large_frags_bytes));
+      size_str(usage->used_by_large_frags_bytes),
+      usage->nlarge_frags_full);
    // clang-format on
 }
 
+/*
+ * Save off shared memory usage stats in a usage struct. This is really
+ * needed to print usage-stats before we dismantle the shared segment
+ * entirely.
+ *
+ * Returns: # of large free-fragments found in-use. Usually when this
+ * function is called for diagnostics, this may not matter. When the
+ * shared segment is dismantled, a non-zero count for "in-use" large fragments
+ * is an indication that something is amiss.
+ */
+int
+platform_save_usage_stats(shminfo_usage_stats *usage, shmem_info *shminfo)
+{
+   *usage                          = shminfo->usage;
+   usage->large_frags_found_in_use = platform_trace_large_frags(shminfo);
```

Comment:
Much simpler rather than the line-by-line copy that old code from L303 onwards was doing.